### PR TITLE
[#2725] Fix(CI): fix CI pipelines do not work as expected

### PR DESCRIPTION
**Pull request by yuqi1129**
### What changes were proposed in this pull request?

Fix the pipeline concurrency configuration and change the cancel conditions.

### Why are the changes needed?

It's a bug to be fixed.

Fix: #2725 

### Does this PR introduce _any_ user-facing change?

N/A.

### How was this patch tested?

Test manually.

### CODE REVIEW

**Comment by mchades** [comment]
Close and reopen for test

**Comment by mchades** [comment]
Overall LGTM

File: .github/workflows/backend-integration-test.yml [hunk 1]
```diff
@@ -9,8 +9,8 @@ on:
     branches: [ "main", "branch-*" ]
 
 concurrency:
-  group: ${{ github.worklfow }}-${{ github.event.pull_request.number || github.ref }}
-  cancel-in-progress: ${{ github.event_name == 'pull_requests' }}
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
```
Comment (jerryshao) [inline .github/workflows/backend-integration-test.yml:13]: What is the purpose of this change?

Comment (yuqi1129) [reply .github/workflows/backend-integration-test.yml:13]: 1. typo: worklfow 
2. There is not event name 'pull_requests', only `pull_request`
3. `cancel-in-progress` means under which event will we cancel pipelines in progress,  there are `push`, `pull_request` and other, `true` means any changes to the PR branch will cancel in process pipelines

**Review by jerryshao — APPROVED** [review]
